### PR TITLE
feat: Remove deprecated syspurpose top-level commands

### PR DESCRIPTION
**Pull request by Glutexo**
Removes _addons_, _role_, _service-level_, and _usage_ commands. These are deprecated as top-level commands as they’ve been moved to the _syspurpose_ command as sub-commands.

[CCT-402](https://issues.redhat.com/browse/CCT-402)

### CODE REVIEW

**Review by jirihnidek — COMMENTED** [review]
You have to at least remove `addons.py` and `test_addons.py`.

**Comment by Glutexo** [comment]
@jirihnidek The [AddonsCommand](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/src/subscription_manager/cli_command/addons.py#L22) class in [addons.py](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/src/subscription_manager/cli_command/addons.py) is [imported](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/src/subscription_manager/cli_command/syspurpose.py#L25) as a [subcommand](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/src/subscription_manager/cli_command/syspurpose.py#L64) in [syspurpose.py](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/src/subscription_manager/cli_command/syspurpose.py).

**Comment by Glutexo** [comment]
Or maybe not. 🤔  This line looks suspicious:

https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/test/cli_command/test_addons.py#L32

But the tests pass even though I removed the command from [managercli](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/src/subscription_manager/managercli.py).

**Comment by Glutexo** [comment]
It works regardless of the argv value… 🤷🏻‍♂️

**Comment by Glutexo** [comment]
I think I found all places where the `addons` command needs to be removed. I‘ll do the same for the other commands moved under the `syspurpose` subcommand.

@jirihnidek Do you think it’s ok this way? Didn’t I forget anything?

**Comment by Glutexo** [comment]
Interesting. In [test_role.py](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/test/cli_command/test_role.py#L17), the [setUp](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/test/cli_command/test_role.py#L17) method doesn’t contain the `argv` [patch](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/test/cli_command/test_addons.py#L32) present in [test_addons.py](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/test/cli_command/test_addons.py). [test_addons.py](https://github.com/candlepin/subscription-manager/blob/dd68e24b5e28bc6469aea87e101422a55edbfb5f/test/cli_command/test_addons.py) however works without the path just fine…

File: src/subscription_manager/cli_command/syspurpose.py
Comment (ptoscano) [inline src/subscription_manager/cli_command/syspurpose.py:41]: if you really want to change this comment in this PR, then simply drop the bit:

| replaced the addons, role, service-level, and usage commands and

once those top-level commands are removed, those references point to removed bits which are hard to find unless you dig into the history; better simply say what the class/command does currently

**Review by ptoscano — CHANGES_REQUESTED** [review]
This seems generally good; please rebase it to fix the conflicts.

Also, if this is ready to be reviewed and merged, please un-draft/un-wip it, thanks.

**Comment by ptoscano** [comment]
This will break the cockpit tests, which currently still use the old commands.
I just created a PR to change those: https://github.com/candlepin/subscription-manager-cockpit/pull/69

This PR will need to wait until that is merged.

(also, is it still "wip"?)

File: src/subscription_manager/cli_command/syspurpose.py
Comment (Glutexo) [reply src/subscription_manager/cli_command/syspurpose.py]: Agreed. Fixed.

**Comment by Glutexo** [comment]
Fine-tuned the pull request, so it can be reviewed and merged:

**Review by ptoscano — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
